packcheck.sh

```diff
@@ -1039,24 +1039,24 @@ find_in_parents () {
 
 ensure_cabal_project() {
   CABALCMD=cabal
-  SDIST_CABALCMD=$CABALCMD
 
   if test -n "$CABAL_PROJECT"
   then
     require_file "$CABAL_PROJECT"
-    SDIST_CABALCMD="$CABALCMD --project-file=$CABAL_PROJECT"
     echo "Using user specified cabal project file at [$CABAL_PROJECT]"
     if test -n "$DISABLE_SDIST_BUILD"
     then # regular build
-      CABALCMD="$SDIST_CABALCMD"
+      SDIST_CABALCMD="$CABALCMD --project-file=$CABAL_PROJECT"
```

Comment:
We use `--project-file` while running sdist build yet don't allow project files while running sdist build.
I'm a little confused.

Reply:
This is only when the user wants to force use of cabal.project by explicitly specifying it via the CABAL_PROJECT env var. Even then we show a warning, in case they did it accidentally.